### PR TITLE
feat: add Prettier

### PR DESCRIPTION
**Pull request by b-mendoza**
Adds Prettier as a code formatter, and Husky and Lint-Staged to run Prettier on the staged files before committing them.

## Motivation and Context

Code consistency.

Fixes https://github.com/rbjks/rbjks.github.io/issues/65

## Description

- A new package.json and a package-lock.json file have been added to the project.
- `concurrently`, `husky`, `lint-staged`, `prettier`, and `prettier-plugin-packagejson` have been added as dev. dependencies.
- A config file for both Prettier and Lint-Staged has been added.
- 4 new scripts have been added to the package.json file to check and fix prettier errors.
- The entire project has been formatted using Prettier.

## How Has This Been Tested?

The changes were tested locally using the [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) extension, and the browser console was checked for any errors.

## Types of Changes

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by agriyakhetarpal** [comment]
Thanks foe the PR, @b-mendoza—it is a bit big but I guess a lot of it is just formatting stuff and related changes—will review this after the new year!

File: CONTRIBUTING.md
Comment (agriyakhetarpal) [inline CONTRIBUTING.md:1]: I was wondering if we can provide any relevant information for future contributors here for configuring Prettier or modifying its usage in the repository as needed. I'm not very well-versed in the JavaScript tooling ecosystem as such, so I would love that too. I hope this doesn't make contributing difficult for newcomers, even though it brings together uniform code quality

**Review by agriyakhetarpal — COMMENTED** [review]
Thanks, @b-mendoza! These changes look pretty great to me and the codebase is indeed _prettier_, just one comment on the instructions for contributing.

File: CONTRIBUTING.md
Comment (adityaraute) [reply CONTRIBUTING.md:1]: I agree.

@b-mendoza would be great if you could help with this.

Comment (b-mendoza) [reply CONTRIBUTING.md:1]: Sounds good to me!

I believe for contributors that use Visual Studio Code, we could add a `.vscode` directory and include a list of recommended extensions to use in the project, one of them being the Prettier extension, which by default will format the code according to the settings of `.prettierrc.js` file.

And by including a list of recommended extensions, VS Code will prompt the user to install them when they open the project for the first time.

<details>
  <summary>Reference of VSCode prompt</summary>

  image
</details>

And for contributors that use a different editor, we could add a section in the `README.md` file with instructions on how to format the code via CLI using Prettier, and the tools you need to have installed to do so.

As in this PR I included as well a couple of scripts to either lint or format the code.

My personal preference would be to add these changes in a new PR, as this one is already quite big.

But let me know what you think 🫡

### References

- https://code.visualstudio.com/docs/editor/extension-marketplace#_recommended-extensions
- https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

Comment (agriyakhetarpal) [reply CONTRIBUTING.md:1]: I agree with you, a follow-up PR would be better for these changes. 

We might go further and add a minimal `.devcontainer` configuration with these recommended extensions as well (though I don't think that will be the roadway for most new contributors).

I am happy to merge this after the conflicts are resolved, unless @adityaraute has any other things to recommend.

Comment (b-mendoza) [reply CONTRIBUTING.md:1]: Hey @agriyakhetarpal 👋 I just fixed the conflicts 🚀

Let me know if I can help with anything else, and I can start working on the next one and add the changes I mentioned in my previous comment.

Comment (b-mendoza) [reply CONTRIBUTING.md:1]: Additionally, I would appreciate if an issue is created to discuss any initial requirements!

**Review by adityaraute — APPROVED** [review]
It looks fine to my ignorant glances. Need your final review before merging it @agriyakhetarpal

**Review by agriyakhetarpal — APPROVED** [review]

**Comment by agriyakhetarpal** [comment]
Please feel free to open an issue where we can discuss any further things to be taken care of @adityaraute (besides the instructions for contributors)

**Comment by adityaraute** [comment]
> Please feel free to open an issue where we can discuss any further things to be taken care of @adityaraute (besides the instructions for contributors)

I thought that's precisely why we need another issue - for the contribution guidelines

**Pull request merged**
